conjure-up/conjure-up
# Checks for availability of LXD

This will check for LXD existence, and update the UI accordingly.

To test:

```
sudo snap remove lxd
sudo apt-get remove -qy lxd lxd-client

conjure-up
```
Once you get to the cloud list you should see localhost greyed out with a
message about how to install snap version of lxd.

next, in another terminal (while leaving conjure-up running)

```
sudo snap install lxd --edge # For now until 2.17 is released.
```

Watch conjure-up update the localhost selection by re-enabling it and allowing
you to deploy to it.

Also, a lot of cleanup removing anything LXD related that we customized
ourselves. This paves the way for the next set of patches that allow the user to
further customize how they want their localhost provider to be configured for
their deployment.

@johnsca please pay closer attention to the async code as I want to make sure it is
keeping with the style of the rest of our async related code.

Fixes #1049

Signed-off-by: Adam Stokes <battlemidget@users.noreply.github.com>
